zulip_bots/zulip_bots/provision.py

```diff
@@ -1,67 +1,85 @@
+
     if os.path.isfile(req_path):
-        print('Installing dependencies...')
+        bot_name = os.path.basename(path_to_bot)
+        logging.info('Installing dependencies for {}...'.format(bot_name))
         if not os.path.isdir(install_path):
             os.makedirs(install_path)
+
         # pip install -r $BASEDIR/requirements.txt -t $BASEDIR/bot_dependencies --quiet
         rcode = pip.main(['install', '-r', req_path, '-t', install_path, '--quiet'])
-        if not rcode == 0:
-            print('Error. Check output of `pip install` above for details.')
+
+        if rcode != 0:
+            logging.error('Error. Check output of `pip install` above for details.')
             if not force:
-                print('Use --force to try running anyway.')
+                logging.error('Use --force to try running anyway.')
                 sys.exit(rcode)  # Use pip's exit code
-        else:
-            print('Installed.')
-        sys.path.insert(0, install_path)
-
-def dir_join(dir1, dir2):
-    # type: (str, str) -> str
-    return os.path.abspath(os.path.join(dir1, dir2))
 
-def run():
-    # type: () -> None
-    usage = '''
-        Installs dependencies of bots in api/bots directory. Add a
-        reuirements.txt file in a bot's folder before provisioning.
-
-        To provision all bots, use:
-        ./provision.py
+        logging.info('Installed dependencies successfully.')
+        sys.path.insert(0, install_path)
 
-        To provision specific bots, use:
-        ./provision.py [names of bots]
-        Example: ./provision.py helloworld xkcd wikipedia
 
-        '''
+def parse_args(available_bots):
+    usage = """
+Installs dependencies of bots in the bots/<bot_name>
+directories. Add a requirements.txt file in a bot's folder
+before provisioning.
 
-    bots_dir = dir_join(os.path.dirname(os.path.abspath(__file__)), 'bots')
-    available_bots = [b for b in os.listdir(bots_dir) if os.path.isdir(dir_join(bots_dir, b))]
+To provision all bots, use:
+./provision.py
 
+To provision specific bots, use:
+./provision.py [names of bots]
+Example: ./provision.py helloworld xkcd wikipedia
+"""
     parser = argparse.ArgumentParser(usage=usage)
+
     parser.add_argument('bots_to_provision',
                         metavar='bots',
                         nargs='*',
                         default=available_bots,
                         help='specific bots to provision (default is all)')
+
     parser.add_argument('--force',
                         default=False,
                         action="store_true",
                         help='Continue installation despite pip errors.')
-    options = parser.parse_args()
+
+    parser.add_argument('--quiet', '-q',
+                        action='store_true',
+                        default=False,
+                        help='Turn off logging output.')
+
+    return parser.parse_args()
+
+
+def main():
+    # type: () -> None
+    bots_dir = os.path.abspath("bots")
```

Comment:
I'm guessing this tool probably doesn't work correctly unless the current working directly is "correct".  Probably a task for a follow-up commit, but we probably want it to find the paths of the files it references based on relative paths.  Scripts in `zulip.git` have lots of examples of doing this correctly.